segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/SegmentMetadata.java

```diff
@@ -204,6 +213,14 @@ public boolean isStorageSystemSegment() {
         return getFlag(StatusFlags.SYSTEM_SEGMENT);
     }
 
+    /**
+     * Gets whether writes are atomic.
+     * @return True if changed, false otherwise.
+     */
+    public boolean isAtomicWrite() {
+        return getFlag(StatusFlags.ATOMIC_WRITES);
```

Comment:
Why is this not covered by tests? Isn't this read somewhere?

Reply:
0.12 doesn't implement atomic writes. So, it doesn't need that flag and doesn't read that flag.

Note that this PR is bare minimum code cherry-picked from 0.13 branch to deserialize the new "append" records and basically ignore them.